kafka/src/main/java/com/expedia/adaptivealerting/kafka/KafkaDetectorManager.java

```diff
@@ -96,11 +106,21 @@ static KafkaDetectorManager buildManager(Config config) {
 
         val metricConsumerConfig = config.getConfig(METRIC_CONSUMER);
         val metricConsumerProps = ConfigUtil.toConsumerConfig(metricConsumerConfig);
-        val metricConsumer = new KafkaConsumer<String, MappedMetricData>(metricConsumerProps);
-        val metricConsumerTopic = metricConsumerConfig.getString(TOPIC);
 
         val anomalyProducerConfig = config.getConfig(ANOMALY_PRODUCER);
         val anomalyProducerProps = ConfigUtil.toProducerConfig(anomalyProducerConfig);
+        if (config.getConfig(DM_TRACING).getString(TRACING_STATUS_STRING).equals(TRACING_STATUS_CHECK_STRING)){
```

Comment:
store config.getConfig(DM_TRACING) in val before if condition and use that inside the if condition to avoid unnecessary call of 
Config tracingDetectorManagerConfig = config.getConfig(DM_TRACING);

Reply:
Fixed.